src/domains/tasks/common/types/index.tsx

```diff
@@ -5,12 +5,18 @@ export const TaskStatusSchema = z.enum([
   "open",
   "assigned",
   "completed",
+] as const);
+
+export type TaskStatus = z.infer<typeof TaskStatusSchema>;
+
+export const PaymentStatusSchema = z.enum([
+  "payment_due",
   "paid",
   "pay_decline",
   "pay_in_processing",
 ] as const);
 
-export type TaskStatus = z.infer<typeof TaskStatusSchema>;
+export type PaymentStatus = z.infer<typeof PaymentStatusSchema>;
 
 export const TaskCategorySchema = z.enum([
```

Comment:
Just to recap, we're defining a Zod schema here so we can validate input.

Since the Zod schema contains the types, can we derive a type from the schema and use that throughout our application.

Reply:
Fixed a typo. That should be "we can derive..." instead of asking "can we derive..."